library/src/blas1/rocblas_copy.hpp

```diff
@@ -0,0 +1,75 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#include "handle.h"
+#include "logging.h"
+#include "rocblas.h"
+#include "utility.h"
+
+template <typename T, typename U, typename V>
+__global__ void copy_kernel(rocblas_int n,
+                            const U     xa,
+                            rocblas_int offsetx,
+                            rocblas_int incx,
+                            rocblas_int stridex,
+                            V           ya,
+                            rocblas_int offsety,
+                            rocblas_int incy,
+                            rocblas_int stridey)
+{
+    ptrdiff_t tid = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    // bound
+    if(tid < n)
+    {
+        const T* x = load_ptr_batch(xa, hipBlockIdx_y, offsetx, stridex);
+        T*       y = load_ptr_batch(ya, hipBlockIdx_y, offsety, stridey);
+
+        if(incx < 0)
+            x -= ptrdiff_t(incx) * (n - 1);
+        if(incy < 0)
+            y -= ptrdiff_t(incy) * (n - 1);
+
+        y[tid * incy] = x[tid * incx];
+    }
+}
+
+template <rocblas_int NB, typename T, typename U, typename V>
+rocblas_status rocblas_copy_template(rocblas_handle handle,
+                                     rocblas_int    n,
+                                     const U        x,
+                                     rocblas_int    offsetx,
+                                     rocblas_int    incx,
+                                     rocblas_int    stridex,
+                                     V              y,
+                                     rocblas_int    offsety,
+                                     rocblas_int    incy,
+                                     rocblas_int    stridey,
+                                     rocblas_stride batch_count)
```

Comment:
We are only using rocblas_stride for stridex and stridey not batch_count so all these need to be flipped.

Reply:
oops, good catch!